docs/architecture/devops-for-aspnet-developers/actions-deploy.md

```diff
@@ -0,0 +1,491 @@
+
+### Create a Repository Secret
+
+1. Now you're going to create an encrypted secret to store the credentials. You'll create this secret at the repository level.
+1. Navigate to GitHub and select your repository **Settings** tab and then select **Secrets**. Select `New repository secret`:
+
+    - ![Create a secret](./media/actions/deploy/add-repository-secret.jpg)
+    **Figure 1**: Create a secret.
+
+1. Copy and paste the JSON from the `az ad sp create-for-rbac` command into the body of the secret. You can create this JSON by hand too if you have the relevant fields for your SPN. The secret should be named `AZURE_CREDENTIALS`. Select `Add secret` to save the new secret:
+
+    - ![Add Azure credentials](./media/actions/deploy/azure-credentials.jpg)
+    **Figure 2**: Add Azure credentials.
+
+1. You'll consume this secret in a workflow in later steps. To access it, you can use the variable notation `${{}}`. In this case, `${{ AZURE_CREDENTIAL }}` will be populated with the JSON you just saved.
+
+## Adding Environments
+
+[Environments](https://docs.github.com/actions/reference/environments) are used as a _logical boundary_. You can add approvals to environments in order to ensure quality. You can also track deployments to environments and specify environment-specific values (secrets) for configuration.
+
+For this example, you're going to split the actual Azure environment into two _logical_ environments called `PRE-PROD` and `PROD`. When you deploy the web app, you'll deploy to the the staging slot of the web application, represented by the `PRE-PROD` environment. When you're ready to deploy to `PROD`, you'll just perform a slot swap.
+
+In this case, the only difference between the environments is the slot that you're deploying to. In real life, there would typically be different web apps (and separate web app plans), separate resource groups and even separate subscriptions. Typically there is an SPN per environment, so you may want to override the `AZURE_CREDENTIAL` value which you saved as a repository secret by creating it as an _environment secret_.
+
+> [!NOTE]
+> Precendence works from Environment to repository. If a targeted environment has a secret called `MY_SECRET`, then that value is used. If not, the repository value of `MY_SECRET` (if any) is used.
+
+1. Select `Settings` and then `Environments` in your repository. Select `New Environment`:
+
+    - ![Create an environment](./media/actions/deploy/new-env.jpg)
+    **Figure 3**: Create an environment.
+
+1. Enter `PRE-PROD` and select `Configure environment`:
+
+    - ![Name the environment](./media/actions/deploy/pre-prod-env.jpg)
+    **Figure 4**: Name the environment.
+
+1. Since deploying to a staging slot does not affect the web app, you can safely deploy to the slot without requiring an approval first. Of course you could add a reviewer if you wanted to! For this example, leave the Environment protection rules empty.
```

Comment:
```suggestion
1. Since deploying to a staging slot doesn't affect the web app, you can safely deploy to the slot without requiring an approval first. A reviewer could be added if desired. For this example, leave the Environment protection rules empty.
```